### PR TITLE
Force WritebackCache on

### PR DESCRIPTION
**Pull request by mystenmark**
Remove PassthroughCache as an option, but leave the code in place so that nodes currently using it do not break

### CODE REVIEW

File: crates/sui-core/src/execution_cache.rs [hunk 1]
```diff
@@ -131,13 +131,11 @@ pub fn choose_execution_cache(config: &ExecutionCacheConfig) -> ExecutionCacheCo
...
     }
 
-    if std::env::var(DISABLE_WRITEBACK_CACHE_ENV_VAR).is_ok()
-        || matches!(config, ExecutionCacheConfig::PassthroughCache)
-    {
-        ExecutionCacheConfigType::PassthroughCache
-    } else {
-        ExecutionCacheConfigType::WritebackCache
+    if std::env::var(DISABLE_WRITEBACK_CACHE_ENV_VAR).is_ok() {
+        error!("DISABLE_WRITEBACK_CACHE is no longer respected. WritebackCache is the default.");
```
Comment (aschran) [inline crates/sui-core/src/execution_cache.rs:135]: should this be fatal?

File: crates/sui-core/src/execution_cache.rs [hunk 2]
```diff
@@ -146,9 +144,19 @@ pub fn build_execution_cache(
     store: &Arc<AuthorityStore>,
 ) -> ExecutionCacheTraitPointers {
     let execution_cache_metrics = Arc::new(ExecutionCacheMetrics::new(prometheus_registry));
-    ExecutionCacheTraitPointers::new(
-        ProxyCache::new(epoch_start_config, store.clone(), execution_cache_metrics).into(),
-    )
+
+    if matches!(
+        epoch_start_config.execution_cache_type(),
+        ExecutionCacheConfigType::WritebackCache
```
Comment (aschran) [inline crates/sui-core/src/execution_cache.rs:150]: would it be safer (future proof) to make this an exhuastive match?

**Review by aschran — APPROVED** [review]

File: crates/sui-core/src/execution_cache.rs (hunk 2 above)
Comment (mystenmark) [reply crates/sui-core/src/execution_cache.rs:149]: there definitely won't be any new cache types - as soon as I can i will strip all of this conditional stuff out.

File: crates/sui-core/src/execution_cache.rs (hunk 1 above)
Comment (mystenmark) [reply crates/sui-core/src/execution_cache.rs:134]: i don't think so? i don't think this will affect anyone anyway.

**Pull request merged**